test/study-webidl.js

```diff
@@ -19,13 +20,32 @@ describe('The Web IDL analyser', async _ => {
     return studyWebIdl(crawlResult);
   }
 
+  function assertEqual(report, ...params) {
+    if (typeof params[0] === 'number') {
+      const idx = params[0];
+      const path = params[1] ?? '';
+      const value = params[2];
+      const actual = _.get(report[idx], path);
+      assert.deepEqual(actual, value,
+        `Expected "${value}" but got "${actual}" while looking at "${path}" in anomaly at index ${idx}. Full anomaly received:\n` +
+        JSON.stringify(report[idx], null, 2));
+    }
+    else {
+      const path = params[0] ?? '';
+      const value = params[1];
+      const actual = _.get(report, path);
+      assert.deepEqual(actual, value,
+        `Expected "${value}" but got "${actual}" while checking report's ${path}. Full report received:\n` +
+        JSON.stringify(report, null, 2));
+    }
+  }
 
   it('reports no anomaly if IDL is valid', async () => {
     const report = await analyzeIdl(`
 [Global=Window,Exposed=*]
 interface Valid {};
 `);
-    assert.deepEqual(report, []);
+    assertEqual(report, []);
```

Comment:
This test passes "by mistake" - in `assertEqual`, both `actual` and `value` ends up being undefined, but no check on the actual content of `report` is done.

Reply:
```suggestion
    assert.deepEqual(report, []);
```

Reply:
Now using a more explicit `assertNbAnomalies(report, 0)`